jetty-websocket/websocket-common/src/main/java/org/eclipse/jetty/websocket/common/message/MessageInputStream.java

```diff
@@ -55,6 +55,7 @@
     {
         RESUMED,
         SUSPENDED,
+        COMPLETE,
```

Comment:
You don't explain why we have a new COMPLETE state, and how it's different then the others.
Eg: what is the different between COMPLETE vs CLOSED?

Reply:
It might be be possible to remove `COMPELTE` now that `close()` does not call resume.
Although it is still useful to enforce no new frames are appended after `fin=true`.